lib/TokenManager.js

```diff
@@ -99,13 +100,16 @@ function get(storage, key) {
 
 function getAsync(sdk, tokenMgmtRef, storage, key) {
   var token = get(storage, key);
-  if (!token || token.expiresAt * 1000 > Date.now()) {
+  var clockSkew = sdk.options.maxClockSkew * 1000;
+  if (!token || token.expiresAt * 1000 > (Date.now() - clockSkew)) {
     return Q.resolve(token);
   }
 
-  return tokenMgmtRef.autoRefresh
+  var tokenPromise = tokenMgmtRef.autoRefresh
     ? refresh(sdk, tokenMgmtRef, storage, key)
     : Q.resolve(remove(tokenMgmtRef, storage, key));
+
+  return new Q(tokenPromise);
```

Comment:
This falls into the same problem as before i.e. an exception thrown in refresh throws an exception (rather than returning a rejected promise).